src/platform/native.zig

```diff
@@ -256,6 +256,21 @@ pub const Platform = struct {
         return platform.last_window_size;
     }
 
+    pub fn hasEvent(platform: *Platform) !bool {
+        try glfw.pollEvents();
+        return platform.events.first != null;
+    }
+
+    pub fn waitEvent(platform: *Platform) !void {
+        _ = platform;
+        try glfw.waitEvents();
+    }
+
+    pub fn waitEventTimeout(platform: *Platform, timeout: f64) !void {
+        _ = platform;
+        try glfw.waitEventsTimeout(timeout);
```

Comment:
Would like to hear @iddev5's thoughts on this PR - I am curious about feasability of implementing in browser.

Reply:
Me too!  I guess if needed it could be implemented by doing short sleeps and checking for events between.

Reply:
waitEvent in general could not be implemented in web. There can be workaround like a canUpdate() function toogled by events which will prevent update() from being called in js.

Reply:
I'm trying to understand how the js version of mach works.  How does wasmUpdate() get called?

My mental model is that it would be called either in response to an input event or a timer.  If that's the case, then waitEvent could return control to the browser and disable the timer.  Can you help me understand?

I'd love to try and get the wasm backend running so I can help there too.  What's the zig build command that you use?

Reply:
Thinking some more, maybe an alternative would be to return info from update() that told the engine whether to wait for an event or to call update() again as soon as possible.  The return type of update() could be !?f64 where null means "call again no waiting" and f64 means "wait for an event with this timeout in seconds"?

Reply:
Js (browsers) can't run an infinite loop because a webpage itself is running with an init-update-deinit loop. So the alternative they provide is a function called ``requestAnimationFrame``. You pass in a callback to this function which appends it to an internal list. At some point in the browser's internal update loop, it pops and executes all the function. This is how update is called for one frame. In order to repeat it for multiple frame, you requestAnimationFrame you update function again as soon as the first one is done. Pseudo code:
```
function update() {
    wasmUpdate();
    requestAnimationFrame(update);
}
requestAnimationFrame(update);
``` 
(Its not recursion)

The code is here in mach: https://github.com/hexops/mach/blob/main/www/template.html#L27-L33

---

Run a wasm application using ``zig build run-example-* -Dtarget=wasm32-freestanding-none``, but the slight problem is that since we dont support webgpu, we get weird integer truncation errors from mach-gpu. So the solution is to make a custom simple example which doesnt do any gpu calls.

Reply:
Thanks very much for the insight!  I'm digging into it.